False positive for `signature-differs` when collecting all arguments

This is a very similar bug to #1553. The main difference is that in this case we set a default value to one of the parameters in the function that we override in the subclass (see ``elit=None``).

### Steps to reproduce
Create a file ``lorem.py`` as follows:

```
class Ipsum:
    def dolor(self, elit=None):
        pass


class LoremIpsum(Ipsum):
    def dolor(self, *args, **kwargs):
        super().dolor(*args, **kwargs)
```

Note that this is a particularly common pattern in Django, for example when overriding the ``save`` method of models (see an [example in the official documentation](https://docs.djangoproject.com/en/3.0/topics/db/models/#overriding-model-methods)).

### Current behavior
Run pylint on the file (``pylint lorem.py``):

```
...
lorem.py:7:4: W0222: Signature differs from overridden 'dolor' method (signature-differs)
```

### Expected behavior
There is no error reported.

### pylint --version output
```
pylint 2.5.3
astroid 2.4.2
Python 3.8.2 (default, Apr 27 2020, 15:53:34) 
[GCC 9.3.0]
```

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -433,3 +433,5 @@ contributors:
 * Raphael Gaschignard: contributor
 
 * Sorin Sbarnea: contributor
+
+* Gergely Kalmár: contributor

--- a/ChangeLog
+++ b/ChangeLog
@@ -83,6 +83,10 @@ Release date: TBA
 
 * Removed incorrect deprecation of ``inspect.getfullargspec``
 
+* Fix ``signature-differs`` false positive for functions with variadics
+
+  Close #3737
+
 What's New in Pylint 2.6.0?
 ===========================
 

--- a/pylint/checkers/classes.py
+++ b/pylint/checkers/classes.py
@@ -1724,7 +1724,10 @@ a metaclass class method.",
             self.add_message(
                 "arguments-differ", args=(class_type, method1.name), node=method1
             )
-        elif len(method1.args.defaults) < len(refmethod.args.defaults):
+        elif (
+            len(method1.args.defaults) < len(refmethod.args.defaults)
+            and not method1.args.vararg
+        ):
             self.add_message(
                 "signature-differs", args=(class_type, method1.name), node=method1
             )

--- a/tests/functional/s/signature_differs.py
+++ b/tests/functional/s/signature_differs.py
@@ -20,3 +20,13 @@ class Cdef(Abcd):
 
     def abcd(self, aaa, bbbb=None): # [signature-differs]
         return aaa, bbbb
+
+
+class Ghij(Abcd):
+    def __init__(self, aaa):
+        Abcd.__init__(self)
+        self.aaa = aaa
+
+    def abcd(self, *args, **kwargs):
+        """Test that a method with variadics does not trigger the warning"""
+        return super().abcd(*args, **kwargs)

